### DISCUSSION TITLE
Ability to bind to 2nd ethernet interface / crashes if eth0 is not up.

### DISCUSSION THREAD

**Discussion by davidsanner** [librespot-org/librespot#622] [Feature Requests]
I'm running librespot (via raspotify) on an RPi 4 running buster w/ a 2nd (usb) ethernet interface (eth1).

It seems librespot is only binding to the other (eth0) interface though the netstat tables says it is listening on 0.0.0.0.  In fact librespot crashes starting up if eth0 is not enabled/up.

Is there and option to allow/deny interfaces or bind to a specific interface or can this be accomplished via an avahi-daemon config?

**Comment by fphammerle** [comment]
I have a similar issue. I'd like `librespot` to be available via multiple network interfaces.

Apparently, `librespot` joins the multicast group for a single network interface only:
```sh
$ ip maddr show dev eth1 | grep 224.0.0.251
	inet  224.0.0.251
$ ip maddr show dev wlan0 | grep 224.0.0.251
$ ip maddr show dev wlan1 | grep 224.0.0.251
$
```
Thus, `librespot` only receives mdns requests (`_spotify-connect._tcp.local`) on `eth1` but not on the wireless interfaces.

My workaround is running
```sh
$ python3 -c 'import socket, struct, threading; s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM); s.setsockopt(socket.SOL_IP, socket.IP_ADD_MEMBERSHIP, struct.pack("=4s4s", socket.inet_aton("224.0.0.251"), socket.inet_aton("[WLAN0_ADDRESS]"))); threading.Event().wait()'
```
to join the multicast group:
```sh
$ ip maddr show dev wlan0 | grep 224.0.0.251
	inet  224.0.0.251
```

I used `sudo tcpdump -i any udp port 5353` to check if `librespot` responds to mdns requests.

**Reply by fphammerle** [reply]
alternative workaround:
```sh
$ sudo iptables -A PREROUTING -i wlan0 -d 224.0.0.251 -j REDIRECT -m comment --comment "librespot mdns"
```

**Reply by Johannesd3** [reply]
It's better to use avahi/bonjour if available. Just compile ot with feature flag `--with-dns-sd`.
